includes/html/api_functions.inc.php

```diff
@@ -1737,6 +1737,44 @@ function list_oxidized(Illuminate\Http\Request $request)
     return response()->json($return, 200, [], JSON_PRETTY_PRINT);
 }
 
+function get_oxidized_state(Illuminate\Http\Request $request): JsonResponse
+{
+    $hostname = $request->route('hostname');
+    $device_id = ctype_digit($hostname) ? (int)$hostname : getidbyname($hostname);
```

Comment:
Instead of using this legacy code (which causes extra sql queries)
You can just put $hostname directly in DeviceCache::get() (it handles device_id/hostname semantics already)